### ISSUE TITLE
[Issue]: ImportError: cannot import name 'auto' from 'tqdm'

### ISSUE THREAD

**Issue by rekilla** [vladmandic/automatic#678]
### Issue Description

Using VENV: D:\stablediff\automatic\venv
22:20:14-729089 INFO     Python 3.10.6 on Windows
22:20:15-217329 INFO     Version: f4256655 Mon May 1 08:43:02 2023 -0400
22:20:15-889376 INFO     Latest published version: d4a748d758e1f3dd4bdf36c045fc30113043fe50 2023-05-01T16:25:35Z
22:20:15-891377 INFO     Setting environment tuning
22:20:15-894378 INFO     nVidia CUDA toolkit detected
22:20:19-931488 INFO     Torch 2.0.0+cu118
22:20:21-717096 INFO     Torch backend: nVidia CUDA 11.8 cuDNN 8700
22:20:21-720108 INFO     Torch detected GPU: NVIDIA GeForce RTX 3060 Laptop GPU VRAM 6144 Arch (8, 6) Cores 30
22:20:21-722128 INFO     Verifying requirements
22:20:28-599109 INFO     No changes detected: Quick launch active
22:20:34-769972 INFO     Running extension preloading
22:20:35-429082 INFO     Server arguments: []
┌───────────────────────────────────────── Traceback (most recent call last) ─────────────────────────────────────────┐
│ D:\stablediff\automatic\launch.py:105 in <module>                                                                   │
│                                                                                                                     │
│   104         exit(0)                                                                                               │
│ > 105     import webui                                                                                              │
│   106     webui.webui()                                                                                             │
│                                                                                                                     │
│ D:\stablediff\automatic\webui.py:20 in <module>                                                                     │
│                                                                                                                     │
│    19 import torchvision # pylint: disable=W0611,C0411                                                              │
│ >  20 import pytorch_lightning # pytorch_lightning should be imported after torch, but it re-e                      │
│    21 logging.getLogger("xformers").addFilter(lambda record: 'A matching Triton is not availab                      │
│                                                                                                                     │
│ D:\stablediff\automatic\venv\lib\site-packages\pytorch_lightning\__init__.py:35 in <module>                         │
│                                                                                                                     │
│   34 from lightning_fabric.utilities.seed import seed_everything  # noqa: E402                                      │
│ > 35 from pytorch_lightning.callbacks import Callback  # noqa: E402                                                 │
│   36 from pytorch_lightning.core import LightningDataModule, LightningModule  # noqa: E402                          │
│                                                                                                                     │
│ D:\stablediff\automatic\venv\lib\site-packages\pytorch_lightning\callbacks\__init__.py:14 in <module>               │
│                                                                                                                     │
│   13 # limitations under the License.                                                                               │
│ > 14 from pytorch_lightning.callbacks.batch_size_finder import BatchSizeFinder                                      │
│   15 from pytorch_lightning.callbacks.callback import Callback                                                      │
│                                                                                                                     │
│ D:\stablediff\automatic\venv\lib\site-packages\pytorch_lightning\callbacks\batch_size_finder.py:24 in <module>      │
│                                                                                                                     │
│    23 import pytorch_lightning as pl                                                                                │
│ >  24 from pytorch_lightning.callbacks.callback import Callback                                                     │
│    25 from pytorch_lightning.tuner.batch_size_scaling import scale_batch_size                                       │
│                                                                                                                     │
│ D:\stablediff\automatic\venv\lib\site-packages\pytorch_lightning\callbacks\callback.py:25 in <module>               │
│                                                                                                                     │
│    24 import pytorch_lightning as pl                                                                                │
│ >  25 from pytorch_lightning.utilities.types import STEP_OUTPUT                                                     │
│    26                                                                                                               │
│                                                                                                                     │
│ D:\stablediff\automatic\venv\lib\site-packages\pytorch_lightning\utilities\types.py:27 in <module>                  │
│                                                                                                                     │
│    26 from torch.utils.data import DataLoader                                                                       │
│ >  27 from torchmetrics import Metric                                                                               │
│    28 from typing_extensions import Protocol, runtime_checkable                                                     │
│                                                                                                                     │
│ D:\stablediff\automatic\venv\lib\site-packages\torchmetrics\__init__.py:14 in <module>                              │
│                                                                                                                     │
│    13                                                                                                               │
│ >  14 from torchmetrics import functional  # noqa: E402                                                             │
│    15 from torchmetrics.aggregation import CatMetric, MaxMetric, MeanMetric, MinMetric, SumMet                      │
│                                                                                                                     │
│ D:\stablediff\automatic\venv\lib\site-packages\torchmetrics\functional\__init__.py:82 in <module>                   │
│                                                                                                                     │
│    81 from torchmetrics.functional.retrieval.reciprocal_rank import retrieval_reciprocal_rank                       │
│ >  82 from torchmetrics.functional.text.bleu import bleu_score                                                      │
│    83 from torchmetrics.functional.text.cer import char_error_rate                                                  │
│                                                                                                                     │
│ D:\stablediff\automatic\venv\lib\site-packages\torchmetrics\functional\text\__init__.py:30 in <module>              │
│                                                                                                                     │
│   29 if _TRANSFORMERS_AVAILABLE:                                                                                    │
│ > 30     from torchmetrics.functional.text.bert import bert_score  # noqa: F401                                     │
│   31     from torchmetrics.functional.text.infolm import infolm  # noqa: F401                                       │
│                                                                                                                     │
│ D:\stablediff\automatic\venv\lib\site-packages\torchmetrics\functional\text\bert.py:24 in <module>                  │
│                                                                                                                     │
│    23                                                                                                               │
│ >  24 from torchmetrics.functional.text.helper_embedding_metric import (                                            │
│    25     TextDataset,                                                                                              │
│                                                                                                                     │
│ D:\stablediff\automatic\venv\lib\site-packages\torchmetrics\functional\text\helper_embedding_metric.py:26 in        │
│ <module>                                                                                                            │
│                                                                                                                     │
│    25 if _TRANSFORMERS_AVAILABLE:                                                                                   │
│ >  26     from transformers import AutoModelForMaskedLM, AutoTokenizer, PreTrainedModel, PreTr                      │
│    27 else:                                                                                                         │
│                                                                                                                     │
│ D:\stablediff\automatic\venv\lib\site-packages\transformers\__init__.py:30 in <module>                              │
│                                                                                                                     │
│     29 # Check the dependencies satisfy the minimal versions required.                                              │
│ >   30 from . import dependency_versions_check                                                                      │
│     31 from .utils import (                                                                                         │
│                                                                                                                     │
│ D:\stablediff\automatic\venv\lib\site-packages\transformers\dependency_versions_check.py:17 in <module>             │
│                                                                                                                     │
│   16 from .dependency_versions_table import deps                                                                    │
│ > 17 from .utils.versions import require_version, require_version_core                                              │
│   18                                                                                                                │
│                                                                                                                     │
│ D:\stablediff\automatic\venv\lib\site-packages\transformers\utils\__init__.py:34 in <module>                        │
│                                                                                                                     │
│    33 )                                                                                                             │
│ >  34 from .generic import (                                                                                        │
│    35     ContextManagers,                                                                                          │
│                                                                                                                     │
│ D:\stablediff\automatic\venv\lib\site-packages\transformers\utils\generic.py:29 in <module>                         │
│                                                                                                                     │
│    28                                                                                                               │
│ >  29 from .import_utils import is_flax_available, is_tf_available, is_torch_available, is_tor                      │
│    30                                                                                                               │
│                                                                                                                     │
│ D:\stablediff\automatic\venv\lib\site-packages\transformers\utils\import_utils.py:34 in <module>                    │
│                                                                                                                     │
│     33                                                                                                              │
│ >   34 from . import logging                                                                                        │
│     35                                                                                                              │
│                                                                                                                     │
│ D:\stablediff\automatic\venv\lib\site-packages\transformers\utils\logging.py:31 in <module>                         │
│                                                                                                                     │
│    30                                                                                                               │
│ >  31 from tqdm import auto as tqdm_lib                                                                             │
│    32                                                                                                               │
└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
ImportError: cannot import name 'auto' from 'tqdm' (D:\stablediff\automatic\venv\lib\site-packages\tqdm\__init__.py)
Press any key to continue . . .

**Comment by vladmandic** [comment]
Can you do

`pip show tqdm`
`pip install -U tqdm`

**Comment by vladmandic** [comment]
i've pinned required `tqdm` version, should resolve this issue. please update and if issue persists, report here and i'll reopen.

**Comment by rekilla** [comment]
C:\Users\NESHE TEFERA>pip show tqdm
Name: tqdm
Version: 4.65.0
Summary: Fast, Extensible Progress Meter
Home-page: https://tqdm.github.io
Author:
Author-email:
License: MPLv2.0, MIT Licences
Location: C:\Users\NESHE TEFERA\AppData\Local\Programs\Python\Python311\Lib\site-packages
Requires: colorama
Required-by: openai, pinecone-client, spacy, webdriver-manager

C:\Users\NESHE TEFERA>pip install -U tqdm
Requirement already satisfied: tqdm in c:\users\neshe tefera\appdata\local\programs\python\python311\lib\site-packages (4.65.0)
Requirement already satisfied: colorama in c:\users\neshe tefera\appdata\local\programs\python\python311\lib\site-packages (from tqdm) (0.4.6)

**Comment by vladmandic** [comment]
Is the issue still present?

**Comment by rekilla** [comment]
yes maybe I will download it again and try but thank you for the answer

**Comment by vladmandic** [comment]
looks like something is downgrading `tqdm` even after installer installs correct version.
which extensions do you have?
also, when error occurs, can you run this?

        venv\Scripts\activate
        python -c "import tqdm; print(tqdm.__version__, tqdm.__file__)"
        python -c "from tqdm import auto; print(auto)"

**Comment by kldzj** [comment]
> which extensions do you have?

After removing all the existing extensions it's starting again, thanks for the hint!

**Comment by vladmandic** [comment]
i'd really love to know which extension messes with core requirements - no extension should do that.
note that specific version of tqdm is required by `transformers` package, its not even specific to this repo.

this should be reported to extension author as critical item.

anyhow, i'm closing the issue as resolved, but it really needs a follow-up with extension itself.

**Comment by kldzj** [comment]
> i'd really love to know which extension messes with core requirements

If it helps, I had deforum, text2video and controlnet installed. The first are the most likely to have been outdated.

**Comment by rekilla** [comment]
> looks like something is downgrading `tqdm` even after installer installs correct version. which extensions do you have? also, when error occurs, can you run this?
> 
> ```
>     venv\Scripts\activate
>     python -c "import tqdm; print(tqdm.__version__, tqdm.__file__)"
>     python -c "from tqdm import auto; print(auto)"
> ```

--------------------------------------------------------------------------------------------------------------

C:\Windows\System32>venv\Scripts\activate

(venv) C:\Windows\System32>python -c "import tqdm; print(tqdm.__version__, tqdm.__file__)"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'tqdm'

(venv) C:\Windows\System32>python -c "from tqdm import auto; print(auto)"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'tqdm'

**Comment by digitalsignalperson** [comment]
I ran into this after installing controlnet and temporalkit

**Comment by vladmandic** [comment]
And why would you install controlnet when it's already installed???

**Comment by digitalsignalperson** [comment]
Sorry I'm not sure what you mean (I'm also fairly new to this). In more detail:

I'm using stable-diffusion-webui-docker, installed sd-webui-controlnet and TemporalKit from the Extensions tab, and it worked until I shutdown the docker container, then trying to start up again would result in the tqdm error message. 

I deleted everything to start fresh and then installed a couple different extensions (sd-webui-controlnet and SD-CN-Animation) and haven't run into the same issue.

**Comment by vladmandic** [comment]
Controlnet is pre-installed here, you installing it once again can only cause problems.

**Comment by digitalsignalperson** [comment]
ohhhh this is a different fork, sorry for the confusion! I'm using https://github.com/AbdBarho/stable-diffusion-webui-docker and https://github.com/AUTOMATIC1111/stable-diffusion-webui
but maybe I will try this fork now :)

**Comment by kldzj** [comment]
> And why would you install controlnet when it's already installed???

It's not documented in your README. If someone is coming from AUTOMATICs SD UI and keeps the extensions this becomes an issue.

Perhaps, as you know which extensions are pre-installed, you might want to check on startup for the existence of such extensions, not load them and warn the user.

**Comment by vladmandic** [comment]
i'll update the readme.
and there is some check already performed - look at console log during server startup.

**Comment by RYG81** [comment]
I removed all extensions still getting error for  import name 'auto' from 'tqdm'

**Comment by vladmandic** [comment]
this issue is already closed, adding to it does not help.
please create new issue and fill all required information. also be careful how you paste logs - look above, that is NOT readable.

**Comment by niekoloid** [comment]
Hi I am also runnning into this problem. Even after removing all the extensions from the extensions folder, I still see ImportError: cannot import name 'auto' from 'tqdm'  error when I run ./webui-user.bat

```
PS C:\Users\x\main\stable-diffusion-webui> pip install tqdm==4.65.0
Requirement already satisfied: tqdm==4.65.0 in c:\users\x\appdata\local\programs\python\python310\lib\site-packages (4.65.0)
Requirement already satisfied: colorama in c:\users\x\appdata\roaming\python\python310\site-packages (from tqdm==4.65.0) (0.4.6)
```

What's the fix for this?

**Comment by alienware377** [comment]
> Hi I am also runnning into this problem. Even after removing all the extensions from the extensions folder, I still see ImportError: cannot import name 'auto' from 'tqdm' error when I run ./webui-user.bat
> 
> ```
> PS C:\Users\x\main\stable-diffusion-webui> pip install tqdm==4.65.0
> Requirement already satisfied: tqdm==4.65.0 in c:\users\x\appdata\local\programs\python\python310\lib\site-packages (4.65.0)
> Requirement already satisfied: colorama in c:\users\x\appdata\roaming\python\python310\site-packages (from tqdm==4.65.0) (0.4.6)
> ```
> 
> What's the fix for this?

after u remove the extension that's causing the issue, delete the venv folder, then start webui again and it will auto repopulate. that should fix the issue.
